main-command/src/main/scala/sbt/Command.scala

```diff
@@ -8,98 +8,155 @@ import sbt.internal.util.complete.{ DefaultParsers, EditDistance, Parser }
+ */
 sealed trait Command {
   def help: State => Help
   def parser: State => Parser[() => State]
+
   def tags: AttributeMap
   def tag[T](key: AttributeKey[T], value: T): Command
 }
-private[sbt] final class SimpleCommand(val name: String, private[sbt] val help0: Help, val parser: State => Parser[() => State], val tags: AttributeMap) extends Command {
-  assert(Command validID name, "'" + name + "' is not a valid command name.")
-  def tag[T](key: AttributeKey[T], value: T): SimpleCommand = new SimpleCommand(name, help0, parser, tags.put(key, value))
+
+private[sbt] final class SimpleCommand(
+    val name: String,
+    private[sbt] val help0: Help,
+    val parser: State => Parser[() => State],
+
+    val tags: AttributeMap
+) extends Command {
+
+  assert(Command validID name, s"'$name' is not a valid command name.")
+
   def help = const(help0)
+
+  def tag[T](key: AttributeKey[T], value: T): SimpleCommand =
+    new SimpleCommand(name, help0, parser, tags.put(key, value))
+
   override def toString = s"SimpleCommand($name)"
 }
-private[sbt] final class ArbitraryCommand(val parser: State => Parser[() => State], val help: State => Help, val tags: AttributeMap) extends Command {
-  def tag[T](key: AttributeKey[T], value: T): ArbitraryCommand = new ArbitraryCommand(parser, help, tags.put(key, value))
+
+private[sbt] final class ArbitraryCommand(
+    val parser: State => Parser[() => State], val help: State => Help, val tags: AttributeMap
+) extends Command {
+  def tag[T](key: AttributeKey[T], value: T): ArbitraryCommand =
+    new ArbitraryCommand(parser, help, tags.put(key, value))
 }
 
 object Command {
   import DefaultParsers._
 
-  def command(name: String, briefHelp: String, detail: String)(f: State => State): Command = command(name, Help(name, (name, briefHelp), detail))(f)
-  def command(name: String, help: Help = Help.empty)(f: State => State): Command = make(name, help)(state => success(() => f(state)))
+  // Lowest-level command construction
+
+  def make(name: String, help: Help = Help.empty)(parser: State => Parser[() => State]): Command =
+    new SimpleCommand(name, help, parser, AttributeMap.empty)
 
   def make(name: String, briefHelp: (String, String), detail: String)(parser: State => Parser[() => State]): Command =
     make(name, Help(name, briefHelp, detail))(parser)
-  def make(name: String, help: Help = Help.empty)(parser: State => Parser[() => State]): Command = new SimpleCommand(name, help, parser, AttributeMap.empty)
 
-  def apply[T](name: String, briefHelp: (String, String), detail: String)(parser: State => Parser[T])(effect: (State, T) => State): Command =
-    apply(name, Help(name, briefHelp, detail))(parser)(effect)
+  // General command construction
+
+  /** Construct a command with the given name, parser and effect. */
   def apply[T](name: String, help: Help = Help.empty)(parser: State => Parser[T])(effect: (State, T) => State): Command =
```

Comment:
Can we add another `apply` method that does not take a parser? I know that's covered by `command`. This would make it easier for people to define commands that take no input from the user, which is a pretty common use case, at least for my sbt workflow.